.github/workflows/tests.yml

```diff
@@ -0,0 +1,201 @@
+        working-directory: build-tests
+        run: |
+          ctest -C Debug --output-on-failure
+
+      - name: Upload Native JARs - Macos 🔺
+        uses: actions/upload-artifact@v3
+        if: runner.os == 'macOS'
+        with:
+          name: libomega_edit.dylib
+          path: lib/libomega_edit.dylib
+
+      - name: Upload Native JARs - Linux 🔺
+        uses: actions/upload-artifact@v3
+        if: runner.os == 'Linux'
+        with:
+          name: libomega_edit.so
+          path: lib/libomega_edit.so
+
+      - name: Upload Native JARs - Windows 🔺
+        uses: actions/upload-artifact@v3
+        if: runner.os == 'Windows'
+        with:
+          name: omega_edit.dll
+          path: lib/omega_edit.dll
+
+  scala-tests:
+    needs: [build-lib-cpp-tests]
+    strategy:
+      matrix:
+        os: [ macos-latest, ubuntu-latest ] # currently don't test windows as it seems to hang
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v3
+
+      - name: Setup Java ☕
+        uses: actions/setup-java@v3.4.1
+        with:
+          distribution: temurin
+          java-version: 8
+          cache: sbt
+
+      - name: Download macos library file 🔻
+        uses: actions/download-artifact@v3
+        if: runner.os == 'macOS'
+        with:
+          name: libomega_edit.dylib
+          path: ./lib/libomega_edit.dylib
+
+      - name: Download linux library file 🔻
+        uses: actions/download-artifact@v3
+        if: runner.os == 'Linux'
+        with:
+          name: libomega_edit.so
+          path: lib/libomega_edit.so
+
+      - name: Download windows library file 🔻
+        uses: actions/download-artifact@v3
+        if: runner.os == 'Windows'
+        with:
+          name: omega_edit.dll
+          path: lib/omega_edit.dll
+
+      - name: Move out library file 🛻
+        run: |
+          if [[ ${{ runner.os }} == 'Linux' ]]; then
+            LIB_FILENAME="libomega_edit.so"
+          elif [[ ${{ runner.os }} == 'macOS' ]]; then
+            LIB_FILENAME="libomega_edit.dylib"
+          else
+            LIB_FILENAME="omega_edit.dll"
+          fi
+
+          mv -v "lib/${LIB_FILENAME}" "lib/${LIB_FILENAME}_dir"
+          mv -v "lib/${LIB_FILENAME}_dir/$LIB_FILENAME" "lib/$LIB_FILENAME"
+          rm -rf "lib/${LIB_FILENAME}_dir"
+        shell: bash
+
+      - name: Check Scala headers
+        run: sbt headerCheckAll
+        working-directory: src/rpc/server/scala
+
+      - name: Package Scala API
+        run: sbt installM2
+        working-directory: src/rpc/server/scala
+
+      - name: Archive M2
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.os }}-artifacts
+          path: ~/.m2/repository/com/ctc/*
+          if-no-files-found: error
+
+      - name: Coverage report
+        run: sbt howMuchCoverage
+        working-directory: src/rpc/server/scala
+
+      - name: Test Scala API
+        run: sbt test
```

Comment:
Another test we ought to run is `sbt serv/test`, which I think we should also run in addition to this API test.